game/src/Game/GameSession/billboard.gdshader

```diff
@@ -0,0 +1,39 @@
+shader_type spatial;
+render_mode unshaded;
+
+//vic2 only ever loads a max of 12 BillboardType
+//Of these, only 4 are actually used
+uniform sampler2D billboards[12] : source_color;
+uniform int numframes[12];
+uniform float sizes[12];
+
+uniform float height_add_factor = 0.3;
+
+//COLOR/INSTANCE_CUSTOM is our custom data, used as follows:
+// x=image index
+// y=frame in image index
+// z,w are unused for now
+
+void vertex() {
+	COLOR = INSTANCE_CUSTOM; //send instance_custom info to fragment
+	float size = sizes[int(COLOR.x)];
+	
+	vec3 cam_right_worldspace = vec3(VIEW_MATRIX[0][0],VIEW_MATRIX[1][0],VIEW_MATRIX[2][0]);
+	vec3 cam_up_worldspace = vec3(VIEW_MATRIX[0][1],VIEW_MATRIX[1][1],VIEW_MATRIX[2][1]);
+	vec3 vert_pos_world =
+		cam_right_worldspace * VERTEX.x * size +
+		cam_up_worldspace * VERTEX.y * size;
+	VERTEX = vert_pos_world + vec3(0,height_add_factor*size,0);
```

Comment:
```suggestion
	VERTEX =
		cam_right_worldspace * VERTEX.x +
		cam_up_worldspace * VERTEX.y;
	VERTEX.y += height_add_factor;
	VERTEX *= size;
```
I think you can reduce the number of operations needed by rearranging stuff like this (although I haven't tested it fully, so I may be wrong somewhere).
Also is it possible to represent the stuff with `cam_right_worldspace` and `cam_up_worldspace` using matrix multiplcation? Just wondering as I'm sure that will be especially fast for the GPU, even if we're ignoring one of the axes by setting `VERTEX.z` to `0`

Reply:
Seems to work.